### DISCUSSION TITLE
Unable to delete users from supabase

### DISCUSSION THREAD

**Discussion by yashsinha-aero** [mckaywrigley/chatbot-ui#1300] [Help]
Keeps saying error deleting user, supabase database error.... works with other supabase projects.. any idea what might be causing this?

**Comment by spammenotinoz** [comment]
Same, my assumption (perhaps incorrect) is that the user still owns resources in the database\storage, so would need to be deleted\cleaned-up from within the FrontEnd application, but as there is no admin interface, likely requires some scripting\knowledge.

**Comment by xycjscs** [comment]
check this in supabase migration setup file. Keep the same, do not change. Then `supabase db reset` rebuild the database.
project_url TEXT := 'http://supabase_kong_chatbotui:8000';

**Comment by spammenotinoz** [comment]
Thankyou, 
I have set project_url TEXT := 'http://supabase_kong_chatbotui:8000'; within 20240108234540_setup.sql, formerly had the URL from hosted supabase (API screen). On the front-end I use an environment variable to set project_url as the hosted superbase URL (from API screen).
Then I pushed the changes to the Database.
I will google, but how to you issue "supabase db reset" for a hosted DB.

**Reply by xycjscs** [reply]
No idea how to deal with hosted db

**Reply by xycjscs** [reply]
When I use hosted supabase, I create cloud db first and then use cloud urls and keys in the environment, not 127.0.0.1:54321. When operations to db is committed in the migration folder, I will copy the content and execute them at the cloud sql editor.

**Comment by spammenotinoz** [comment]
Any tips for doing the reset when I try locally, even with a fresh install I get,
Applying migration 20240108234545_add_file_items.sql...
unexpected EOF
At statement 0: --------------- FILE ITEMS ---------------

**Reply by xycjscs** [reply]
same problem. But I forgot how to solve this problem.

**Reply by spammenotinoz** [reply]
Managed to reset the database, think it was superbase init "really can't recall, as I got myself in a mess".
Anyway I did a fresh start project_url TEXT := 'http://supabase_kong_chatbotui:8000';, clean database with no users.
But I still can not delete any users, get the "database error".
Using hosted, thinking about pulling the DB down, then trying to delete users locally then push back up.

Hmm.. I still can't delete users.
I am trying to delete users directly out of supabase via the GUI is this the correct way?

**Comment by spammenotinoz** [comment]
Hi @SwiftDev29, were you able to resolve this?  I have not been able to fix, even with a new deployment.

**Reply by yashsinha-aero** [reply]
I tired some of the advice here, replacing the URL of the projectURL seemed to work in a fresh project .

**Comment by spammenotinoz** [comment]
Found this in the SQL logs:
Could not resolve host: supabase_kong_chatbotui
![image](https://github.com/mckaywrigley/chatbot-ui/assets/29272432/a95805a2-e6e8-4d5f-8c13-0f4cb0be17f4)

**Comment by spammenotinoz** [comment]
Okay, the fix was to re-run the storage migration sql with the correct project_url (API settings) and service key.

CREATE OR REPLACE FUNCTION delete_storage_object(bucket TEXT, object TEXT, OUT status INT, OUT content TEXT)
RETURNS RECORD
LANGUAGE 'plpgsql'
SECURITY DEFINER
AS $$
DECLARE
  project_url TEXT := '<take your project ID from API settings';
  service_role_key TEXT := '<replace with your actual service role key';
  url TEXT := project_url || '/storage/v1/object/' || bucket || '/' || object;
BEGIN
  SELECT
      INTO status, content
           result.status::INT, result.content::TEXT
      FROM extensions.http((
    'DELETE',
    url,
    ARRAY[extensions.http_header('authorization','Bearer ' || service_role_key)],
    NULL,
    NULL)::extensions.http_request) AS result;
END;
$$;
